Vof pw v2 (#476)
lethe-cfd/lethe: doc/source/parameters/cfd/volume_of_fluid.rst

    The cell is then filled with the lower density fluid by changing its phase value.

  * Wetting of the lower density fluid occurs where those conditions are met: 

    * the cell is in the domain of the lower density fluid,
    * the cell pressure value is above ``wetting pressure value``, and
    * the distance (on the phase value) to the interface is above ``wetting phase distance``.

    The cell is then filled with the higher density fluid by changing its phase value.

    .. tip::

      For ``set wetting phase distance = 0``, the wetting can only occur at the interface (considered at ``phase value = 0.5``).

      For ``set wetting phase distance`` :math:`> 0`, the wetting can occur in the area where is larger than the area occupied by the higher density fluid. For example:

      * if the ``fluid 1`` has a higher density than ``fluid 0``, and ``set wetting phase distance = 0.1``, the wetting can occur where the phase value is below :math:`= 0.4`.
      * if the ``fluid 0`` has a higher density than ``fluid 1``, and ``set wetting phase distance = 0.1``, the wetting can occur where the phase value is above :math:`= 0.6`.

> * ``diffusion``: value of the diffusion in the transport equation of the phase fraction. Default value is 0 to have pure advection. This can be used to :ref:`improve wetting`.

Author: right now `diffusion` is outside of the peeling wetting subsection. Though I think I will put it inside of it, because I do not see any other case where we would like to add a diffusivity if not to improve the wetting mechanism! ^^

  * ``monitoring``: controls if conservation is monitored at each iteration, through the volume computation of the fluid with index ``fluid monitored``. Results are outputted in a data table (`VOF_monitoring_fluid_0.dat` or `VOF_monitoring_fluid_1.dat`).

  .. admonition:: Example of file output, `VOF_monitoring_fluid_1.dat`:
  
    .. code-block:: text

	 time   volume_fluid_1 
	 0.0000     4.9067e-01 
	 0.0100     4.8425e-01 
	 0.0200     4.8564e-01 
	 0.0300     4.7858e-01 
	 0.0400     4.8245e-01 
	 0.0500     4.7693e-01 
	 0.0600     4.8154e-01 
	 0.0700     4.7590e-01 
	 0.0800     4.8133e-01 
	 0.0900     4.7604e-01 
	 0.1000     4.8198e-01 

.. tip::
  Due to numerical diffusion of the interface, the ``peeling wetting`` mechanism or an added ``diffusion``, the method used is not strictly conservative at every iteration. The ``sharpening threshold`` can then be adapted to counter fluid loss (e.g. ``set sharpening threshold = 0.45``) or creation (e.g. ``set sharpening threshold = 0.55``).

> * ``subsection surface tension force``: Surface tension is the tendency of a liquid to maintain the minimum possible surface area. This subsection defines parameters to ensure an accurate interface between the two phases, used when at least one phase is liquid. 

Reviewer: ```suggestion
* ``subsection surface tension force``: Surface tension is the tendancy of a liquid to maintain the minimum possible surface area. This subsection defines parameters to ensure an accurate interface between the two phases, used when at least one phase is liquid. 
```

  * ``enable``: controls if ``surface tension force`` is considered.
  * ``verbosity``: enables the display of the output from the surface tension force calculations. Choices are: ``quiet`` (default, no output) and ``verbose``.
  * ``output auxiliary fields``: enables the display of the filtered ``phase fraction gradient`` and filtered ``curvature``. Used for debugging purposes.

>   * ``surface tension coefficient``: surface tension coefficient in :math:`Nm^{-1}`.

Reviewer: Maybe there could be an equation for this?

  * ``phase fraction gradient filter``: value used to apply a `projection step <https://onlinelibrary.wiley.com/doi/full/10.1002/fld.2643>`_ to damp high frequency errors, that are magnified by differentiation, in the phase fraction gradient (:math:`\bf{\psi}`), following the equation:

    .. math::

>         \int_\Omega {\bf{v}} . {\bf{\psi}} + \eta_n \nabla {\bf{v}} . \nabla {\bf{\psi}} d\Omega = \int_\Omega {\bf{v}} . \nabla {\phi} d\Omega

Reviewer: ```suggestion
        \int_\Omega ({\bf{v}} \cdot {\bf{\psi}} + \eta_n \nabla {\bf{v}} \cdot \nabla {\bf{\psi}}) d\Omega = \int_\Omega {\bf{v}} \cdot \nabla {\phi} d\Omega
```

    where :math:`\bf{v}` is a piecewise continuous vector-valued test function, :math:`\bf{\psi}` is the filtered phase fraction gradient, :math:`\eta_n \geq 0` is the ``phase fraction gradient filter`` value, and :math:`\phi` is the phase fraction.

  .. tip::

    The ``phase fraction gradient filter`` must be a small value larger than 0. Use the procedure suggested in: :ref:`choosing values for the surface tension force filters`.

  * ``curvature filter``: value used to apply a `projection step <https://onlinelibrary.wiley.com/doi/full/10.1002/fld.2643>`_ to damp high frequency errors, that are magnified by differentiation, in the curvature (:math:`k`), following the equation:

    .. math:: 

>         \int_\Omega v k + \eta_k \nabla v . \nabla k d\Omega = \int_\Omega \nabla v . \frac{\bf{\psi}}{|\bf{\psi}|} d\Omega

Reviewer: ```suggestion
        \int_\Omega (v k + \eta_k \nabla v \cdot \nabla k) d\Omega = \int_\Omega \nabla v \cdot \frac{\bf{\psi}}{|\bf{\psi}|} d\Omega
```

    where :math:`v` is a test function, :math:`k` is the filtered curvature, :math:`\eta_k` is the ``curvature filter`` value, and :math:`\bf{\psi}` is the filtered phase fraction gradient. 

  .. tip::

    Use the procedure suggested in: :ref:`choosing values for the surface tension force filters`.

.. _improve wetting:

Improving the Wetting mechanism
+++++++++++++++++++++++++++++++++++

In the framework of incompressible fluids, a layer of the lowest density fluid (e.g. air) can form between the highest density fluid (e.g. water) and the boundary, preventing its wetting. Two strategies can be used to improve the wetting mechanism:

> 1. add a small ``diffusion`` to the transport equation (e.g. ``set diffusion = 1e-3``), so that the higher density fluid spreads to the boundary location. 

Reviewer: ```suggestion
1. Add a small ``diffusion`` to the transport equation (e.g. ``set diffusion = 1e-3``), so that the higher density fluid spreads to the boundary location. 
```

.. tip::

>   It is strongly advised to sharpen the interface more often (e.g. ``set sharpening frequency = 2``) to limit interface blurriness due the added diffusion. As peeling-wetting is handled after the transport equation is solved, but before interface sharpening, this will not prevent the wetting to occur.

Reviewer: ```suggestion
  It is strongly advised to sharpen the interface more often (e.g. ``set sharpening frequency = 2``) to limit interface blurriness due the added diffusion. As peeling-wetting is handled after the transport equation is solved, but before interface sharpening, this will not prevent the wetting from occuring.
```

> 2. remove the conservation condition on the lowest density fluid (e.g. ``set skip mass conservation in fluid 0 = false``). The mass conservation equation in the cells of interest is replaced by a zero-pressure condition, to allow the fluid to get out of the domain. 

Reviewer: ```suggestion
2. Remove the conservation condition on the lowest density fluid (e.g. ``set skip mass conservation in fluid 0 = false``). The mass conservation equation in the cells of interest is replaced by a zero-pressure condition, to allow the fluid to get out of the domain. 
```

.. tip::
  This can give more precise results as the interface remains sharp, but the time step (in :doc:`simulation_control`) must be low enough to prevent numerical instabilities.

.. _choosing values for the surface tension force filters:

Choosing values for the surface tension force filters
+++++++++++++++++++++++++++++++++++++++++++++++++++++++

The following procedure is recommended to choose proper values for the ``phase fraction gradient filter`` and ``curvature filter``: 

> 1. use ``set output auxiliary fields = true`` 
> 2. choose a small value, still larger than :math:`0`, for example :math:`h/10` with :math:`h` the smallest mesh size 
> 3. run the simulation and check whether the filtered phase fraction gradient field is smooth and without oscillation 
> 4. if the filtered field (``phase fraction gradient`` or ``curvature``) shows oscillations, increase the value, for example :math:`h/5`, and repeat this process until reaching a smooth filtered field without oscillations.

Reviewer: ```suggestion
1. Use ``set output auxiliary fields = true`` 
2. Choose a small value, still larger than :math:`0`, for example :math:`h/10` with :math:`h` the smallest mesh size 
3. Run the simulation and check whether the filtered phase fraction gradient field is smooth and without oscillation 
4. If the filtered field (``phase fraction gradient`` or ``curvature``) shows oscillations, increase the value, for example :math:`h/5`, and repeat this process until reaching a smooth filtered field without oscillations.
```

	 0.0900     4.7604e-01 
	 0.1000     4.8198e-01 

.. _tip on sharpening threshold:

.. tip::
  Due to numerical diffusion of the interface, the ``peeling wetting`` mechanism or an added ``diffusivity``, the method used is not strictly conservative at every iteration. The ``sharpening threshold`` can then be adapted to counter fluid loss (e.g. ``set sharpening threshold = 0.45``) or creation (e.g. ``set sharpening threshold = 0.55``).

> * ``subsection surface tension force``: Surface tension is the tendancy of a liquid to maintain the minimum possible surface area. This subsection defines parameters to ensure an accurate interface between the two phases, used when at least one phase is liquid. 

Reviewer 2: I think tendency is in english and tendance is in french
